cc/layers/layer.cc

```diff
@@ -119,14 +129,9 @@ Layer::Layer()
       property_tree_sequence_number_(-1),
       ignore_set_needs_commit_for_test_(false),
       bitflags_(0u),
-      subtree_property_changed_(false) {
-  // https://linear.app/replay/issue/RUN-885
-  recordreplay::RegisterPointer("Layer", this);
```

Comment:
I'm less sure about removing the existing registration here.  There may be other places / asserts where we use `GetPointerId()` to map the raw pointer to a stable id.  But other than that this looks ok.

Reply:
We don't really have a developed strategy to make sure we keep track of all the places where PointerId() is used on different types and makes it easy to reason about removing unnecessary RegisterPointer calls.  There is a convention of adding comments pointing to the usage or issues which the RegisterPointer is needed for, but that isn't always going to be present or guaranteed to be up to date.

If there are still uses of PointerId(), they could lead either to bogus mismatches that test the pointer ID, or crashes when CompareByPointerId or a similar class is used to sort objects by their pointer ID.  The former case won't cause crashes for users but will show up in crash reports and can be fixed by there, the latter case should be caught by the testing we do in CI before releasing a build.

In this case I think it should be fine to remove this RegisterPointer call.